response.go

```diff
@@ -1,25 +1,75 @@
 package freshchat
 
-type freshchatResponse struct {
-	success        *successResponse
-	failed         *failedResponse
-	httpStatusCode int
-	rawData        string
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+)
+
+// ResponseFreshchat is the response from Freshchat.
+type ResponseFreshchat struct {
+	Success        *ResponseSuccess `json:"-"`
+	Failed         *ResponseFailed  `json:"-"`
+	HTTPStatusCode int              `json:"-"`
+	RawData        string           `json:"-"`
 }
 
-type successResponse struct {
-	RequestId          string `json:"request_id"`
-	RequestProcessTime string `json:"request_process_time"`
-	Link               link   `json:"link"`
+// assign assigns the values from the given *http.Response.
+func (r *ResponseFreshchat) assign(resp *http.Response) (res *ResponseFreshchat, err error) {
+	if resp == nil || resp.Body == nil {
+		return
+	}
+
+	byteBody, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return
+	}
+
+	var ptrDecode interface{}
+	if resp.StatusCode >= http.StatusBadRequest {
+		ptrDecode = &r.Failed
+	} else {
+		ptrDecode = &r.Success
+	}
+
+	err = json.Unmarshal(byteBody, ptrDecode)
+	if err != nil {
+		return
+	}
+
+	defer func() {
+		res = r
+	}()
+	r.HTTPStatusCode = resp.StatusCode
+	r.RawData = convertByteToString(byteBody)
+	err = r.getError()
+	return
 }
 
-type link struct {
-	Rel  string `json:"rel"`
+func (r *ResponseFreshchat) getError() (err error) {
+	if !(r.HTTPStatusCode >= http.StatusBadRequest) {
+		return
+	}
+
+	return r.Failed
```

Comment:
My opinion, can we change the flow validation like this mas, so we can easy to read if condition.
```
if r.HTTPStatusCode >= http.StatusBadRequest {
    return r.Failed
}

return
```

Reply:
@asqolani-flip Ok, I have changed it. Please check.